sopel/coretasks.py

```diff
@@ -1549,7 +1549,11 @@ def _record_who(
         }
         for c in modes:
             priv = priv | mapping[c]
-    if channel not in bot.channels:
+
+    # `*` placeholder is returned for users with no visible channels; see #2675
+    if channel == '*':
+        return
+    elif channel not in bot.channels:
```

Comment:
Since you are using `return` here you can simplify the if/elif (since you don't have a generic `else`) :

```suggestion
    if channel == '*':
        return

    if channel not in bot.channels:
```

Reply:
Also, you can put that `if *: return` before the `if modes` above!